test/document_stores/test_faiss.py

```diff
@@ -8,294 +9,266 @@
+        ds.write_documents(documents_embedding_only, index="index2")
+
+        docs_from_index1 = ds.get_all_documents(index="index1", return_embedding=False)
+        assert len(docs_from_index1) == len(documents_embedding_only)
+        assert {int(doc.meta["vector_id"]) for doc in docs_from_index1} == set(range(0, 6))
+
+        docs_from_index2 = ds.get_all_documents(index="index2", return_embedding=False)
+        assert len(docs_from_index2) == len(documents_embedding_only)
+        assert {int(doc.meta["vector_id"]) for doc in docs_from_index2} == set(range(0, 6))
+
+    @pytest.mark.integration
+    def test_update_docs_different_indexes(self, ds, documents_embedding_only):
+        retriever = MockDenseRetriever(document_store=ds)
+
+        ds.write_documents(documents_embedding_only, index="index1")
+        ds.write_documents(documents_embedding_only, index="index2")
+
+        ds.update_embeddings(retriever=retriever, update_existing_embeddings=True, index="index1")
+        ds.update_embeddings(retriever=retriever, update_existing_embeddings=True, index="index2")
+
+        docs_from_index1 = ds.get_all_documents(index="index1", return_embedding=False)
+        assert len(docs_from_index1) == len(documents_embedding_only)
+        assert {int(doc.meta["vector_id"]) for doc in docs_from_index1} == set(range(0, 6))
+
+        docs_from_index2 = ds.get_all_documents(index="index2", return_embedding=False)
+        assert len(docs_from_index2) == len(documents_embedding_only)
+        assert {int(doc.meta["vector_id"]) for doc in docs_from_index2} == set(range(0, 6))
+
+    @pytest.mark.integration
+    @pytest.mark.parametrize("index_factory", ["Flat", "HNSW", "IVF1,Flat"])
+    def test_retrieving(self, documents_embedding_only, index_factory, tmp_path):
+        document_store = FAISSDocumentStore(
+            sql_url=f"sqlite:///{tmp_path}/test_faiss_retrieving_{index_factory}.db",
+            faiss_index_factory_str=index_factory,
+            isolation_level="AUTOCOMMIT",
+        )
 
+        document_store.delete_all_documents(index="document")
+        if "ivf" in index_factory.lower():
+            document_store.train_index(documents_embedding_only)
+        document_store.write_documents(documents_embedding_only)
 
-@pytest.mark.skipif(sys.platform in ["win32", "cygwin"], reason="Test with tmp_path not working on windows runner")
-def test_faiss_passing_index_from_outside(tmp_path):
-    d = 768
-    nlist = 2
-    quantizer = faiss.IndexFlatIP(d)
-    index = "haystack_test_1"
-    faiss_index = faiss.IndexIVFFlat(quantizer, d, nlist, faiss.METRIC_INNER_PRODUCT)
-    faiss_index.set_direct_map_type(faiss.DirectMap.Hashtable)
-    faiss_index.nprobe = 2
-    document_store = FAISSDocumentStore(
-        sql_url=f"sqlite:////{tmp_path/'haystack_test_faiss.db'}",
-        faiss_index=faiss_index,
-        index=index,
-        isolation_level="AUTOCOMMIT",
-    )
+        retriever = EmbeddingRetriever(
```

Comment:
Let's use a mock retriever, the output is not tested in any way

Reply:
In the end that test was only exercising the indexing, I removed it altogether, moving the index factory parameter to the write documents test